### DISCUSSION TITLE
SQLAlchemy 2.0 ORM - how do I insert a value into a non-nullable column?

### DISCUSSION THREAD

**Discussion by moonman239** [sqlalchemy/sqlalchemy#8422] [Usage Questions]
Hello all!

I have a class that looks like this:

`class CustomORM(Base):
    id = Column(Integer,primary_key=True)
    col1 = Column(String, nullable=False)
    col2 = Column(Integer,nullable=False)
    col3 = Column(Integer,nullable = True)
`

I want to know how to properly instantiate an object of this class so that I can add it later with session.add.

The documentation (https://docs.sqlalchemy.org/en/20/tutorial/orm_data_manipulation.html) doesn't seem to be clear on how to insert objects with non-nullable columns, thus my confusion.

How do I do what I'm looking to do?

**Comment by zzzeek** [comment]
you would instantiate it exactly as you see in the document you linked at https://docs.sqlalchemy.org/en/20/tutorial/orm_data_manipulation.html#instances-of-classes-represent-rows , are you getting an error of some kind?

**Reply by moonman239** [reply]
no, just wasn't sure if the syntax was any different.
I thought kwargs were supposed to be optional, but I guess it's easiest for the SQLAlchemy devs to use them anyway.

**Reply by zzzeek** [reply]
they are optional.  database fields are often assigned from the server as well, such as integer primary key values.  additionally, the object is not INSERTed when you construct it.  you might want to set the fields on it later.

that said, SQLA 2.0 offers a new constructor feature that will do the non-optional positional arguments as well by mapping as a dataclass, see https://docs.sqlalchemy.org/en/20/orm/dataclasses.html#declarative-dataclass-mapping
